File: app/src/main/java/com/lucasurbas/counter/service/Timers.java
package com.lucasurbas.counter.service;

public interface Timers {

    interface TimerListener {

        void onTick(long millisUntilFinished);

        void onFinish();
    }

    void startNew(int timerId, long millisInFuture, TimerListener listener);

    void remove(int timerId);

    boolean hasTimer(int timerId);
}

File: app/src/main/java/com/lucasurbas/counter/service/Timers.kt
package com.lucasurbas.counter.service

interface Timers {

    interface TimerListener {

        fun onTick(millisUntilFinished: Long)

        fun onFinish()
    }

    fun startNew(timerId: Int, millisInFuture: Long, listener: TimerListener)

    fun remove(timerId: Int)

    fun hasTimer(timerId: Int): Boolean
}
